fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -112,10 +139,12 @@ public void primaryIndexDoesOverlapPrimaryKey(final TestHelpers.BooleanEnum depr
         assertArrayEquals(new int[] {0}, index.getPrimaryKeyComponentPositions());
```

Comment:
It's kind of weird that the counter-based key was added to this method but then the value isn't checked. It seems like, in general, the subspace key should be checked if the counter-based mechanism is added to a test or it shouldn't be added at all.